src/attributes/classes.js

```diff
@@ -158,12 +156,13 @@ jQuery.fn.extend({
 	},
 
 	hasClass: function( selector ) {
-		var className = " " + selector + " ",
-			i = 0,
-			l = this.length;
-		for ( ; i < l; i++ ) {
-			if ( this[i].nodeType === 1 &&
-				( " " + getClass( this[i] ) + " " ).replace( rclass, " " )
+		var className, elem,
+			i = 0;
+
+		className = " " + selector + " ";
+		while ( ( elem = this[ i++ ] ) ) {
+			if ( elem.nodeType === 1 &&
+				( " " + getClass( elem ) + " " ).replace( rclass, " " )
```

Comment:
The combined boolean and string assignment and repeated check of `cur` is a bit odd though. Perhaps like this:

```js
hasClass: function( selector ) {
    var cur, elem,
        i = 0;

    selector = " " + selector + " ";
    while ( ( elem = this[ i++ ] ) ) {
        if ( elem.nodeType === 1 ) {
            cur = ( " " + getClass( elem ) + " " ).replace( rclass, " " );
            if ( cur.indexOf( selector ) > -1 ) {
                return true;
            }
        }
    }

    return false;
}
```

It seems UglifyJS optimises this quite well. It produces a `for` statement for the setup, and a single `if` statement as its body.